include/rocksdb/env.h

```diff
@@ -69,6 +69,12 @@ struct ConfigOptions;
 
 const size_t kDefaultPageSize = 4 * 1024;
 
+// This type definition is for transitioning to 7.0 release and will
+// be removed soon after. It will be changed to = uint64_t in 7.0 so
+// that APIs consistently use uint64_t for representing file offsets
+// and sizes that can be beyond memory size.
+using file_size_t = size_t;
```

Comment:
We should be really careful in creating alias in header file (also see Google C++ Style https://google.github.io/styleguide/cppguide.html#Aliases), especially a name that is so common. If we have to do it, naming is something like rocksdb_file_size_t to prevent possible conflict with users' alias or variable name.

Reply:
This is within namespace, so perhaps conflict is not the problem, but still, is it something that we hope users to see in public header?

Reply:
Are you concerned about cases where people have `using namespace rocksdb`? That's the only case where there'd be ambiguity.

Reply:
> is it something that we hope users to see in public header?

Users can take advantage of it if they choose, keeping in mind "will be removed soon after" 7.0. It doesn't work for its intended purposes unless it's in the public header.